dpnp/backend/rng_engine.hpp

```diff
@@ -0,0 +1,90 @@
+//*****************************************************************************
+// Copyright (c) 2016-2020, Intel Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+// - Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the following disclaimer.
+// - Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//*****************************************************************************
+
+#pragma once
+#ifndef RNG_ENGINE_H
+#define RNG_ENGINE_H
+
+#include <iostream>
+
+#include <ctime>
+#include <mkl_sycl.hpp>
+
+#include "backend_utils.hpp"
+#include "queue_sycl.hpp"
+
+namespace mkl_rng = oneapi::mkl::rng;
+
+#define DPNP_RNG_ENGINE engine_rng::get_engine()
+
+/**
+ * This is
+ * TODO:
+ * add docs
+ */
+
+class engine_rng
+{
+    // TODO:
+    // static
+    static mkl_rng::mt19937* mt19937_engine;

```

Comment:
move this definition to `class backend_sycl`
https://github.com/IntelPython/dpnp/blob/master/dpnp/backend/queue_sycl.hpp#L43